src/frontend/components/Chat/SharedChatComponents/ChatConversationDisplayer/index.tsx

```diff
@@ -30,26 +32,24 @@ export const ChatConversationDisplayer = () => {
         top: scrollableContainerRef.current.scrollHeight,
       });
     }
-  }, [messages]);
+  }, [messages, presenceItems]);
 
   return (
     <Box
+      height="100%"
       overflow={{
-        vertical: 'auto',
         horizontal: 'hidden',
+        vertical: 'auto',
       }}
       ref={scrollableContainerRef}
-      height="100%"
     >
       {loading ? (
         <Box align="center" height="100%" justify="center" width="100%">
           <Spinner size="large" />
         </Box>
       ) : (
         <React.Fragment>
-          {messages.map((msg, idx) => (
-            <ChatMessageItem key={idx} msg={msg} />
-          ))}
+          {computeChatItems(messages, presenceItems)}
```

Comment:
This function can be heavy with lot of messages and presences. What about using a useMemo here ?

Reply:
I changed how the ChatConversationDisplayer component works, in order to address the performance issue you were mentioning here.
Now all computed components (messages and presences) are stored in a state in ChatConversationDisplayer. When a new message or a new presence is received by converse, it is processed alone and then add to the state.
Also, messages received from history are first added to the chat and only then presences can be added (indeed, there is no history for presences, so new presences received will necessarily be after history messages).